### PR TITLE
Warning improvements

### PR DESCRIPTION
**Pull request by kripken**
* If `wasm-opt` is run with no passes, warn, as we've gotten reports that people assume a tool called "wasm-opt" should optimize automatically (but we follow llvm's `opt` convention of not doing so).
 * Add a `--quiet` flag that suppresses this minor warning, and the other minor warning where there is no output file.

### CODE REVIEW

File: test/unit/test_warnings.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import subprocess
+import tempfile
+
+from scripts.test.shared import WASM_OPT, WASM_DIS, WASM_SHELL, NODEJS, run_process
+from .utils import BinaryenTestCase
+
+
+class WarningsText(BinaryenTestCase):
+    def test_warn_on_no_passes(self):
+      err = run_process(WASM_OPT + [self.input_path('asyncify-pure.wast'), '-o', 'a.wasm'], stderr=subprocess.PIPE).stderr
```
Comment (tlively) [inline test/unit/test_warnings.py]: The indentation isn't consistent here.

Comment (kripken) [reply test/unit/test_warnings.py]: Sorry, what isn't consistent? Your eyes must be better than mine :)

Comment (kripken) [reply test/unit/test_warnings.py]: Oh, 4 spaces... fixing.

File: test/unit/test_warnings.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import os
```
Comment (sbc100) [inline test/unit/test_warnings.py]: Add copyright header

Comment (kripken) [reply test/unit/test_warnings.py]: Looks like none of the test files in `test/unit/` have copyright headers. Should they? (does test code need them in general?)

Comment (sbc100) [reply test/unit/test_warnings.py]: In general yes, but I guess we can do that in bulk as as separate change.

**Pull request merged**